vms/platformvm/block/builder/builder.go

```diff
@@ -173,7 +177,7 @@ func (b *builder) ResetBlockTimer() {
 	b.timer.SetTimeoutIn(0)
 }
 
-func (b *builder) setNextBuildBlockTime() {
+func (b *builder) maybeAdvanceTime() {
```

Comment:
`maybeAdvanceTime` can be called due to changes in preference. This can cause the next staker change time to increase _or decrease_. So I don't think this function is currently sufficient.

Reply:
re-worked `ResetBlockTimer` to update `nextStakerChangeTime`